tooling/prometheus-rules/main.go

```diff
@@ -72,11 +68,62 @@ func main() {
 	if err := o.complete(flag.Args()); err != nil {
 		logrus.WithError(err).Fatal("could not complete options")
 	}
-	if err := generate(o.rules, o.output); err != nil {
+	b, _ := os.ReadFile("testing-prometheusRule_test.yaml")
+	if err := runTests(o.inputRules, o.rules, b); err != nil {
+		logrus.WithError(err).Fatal("testing rules failed")
+	}
+	output, err := os.Create(o.outputBicep)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to create output file")
+	}
+	if err := generate(o.rules, output); err != nil {
 		logrus.WithError(err).Fatal("failed to generate bicep")
 	}
 }
 
+func runTests(inputRulesFile string, input monitoringv1.PrometheusRule, testFileContent []byte) error {
+	dir, err := os.MkdirTemp("/tmp", "prom-rule-test")
+	if err != nil {
+		return fmt.Errorf("Error creating tempdir %v", err)
+	}
+	defer func() {
+		os.RemoveAll(dir)
+	}()
+
+	logrus.Infof("Created tempdir %s", dir)
+
+	ruleGroups, err := yaml.Marshal(input.Spec)
+	if err != nil {
+		return fmt.Errorf("error Marshalling rule groups %v", err)
+	}
+
+	tmpFile := fmt.Sprintf("%s%s%s", dir, string(os.PathSeparator), inputRulesFile)
+
+	err = os.WriteFile(tmpFile, ruleGroups, 0644)
+	if err != nil {
+		return fmt.Errorf("error writing rule groups file %v", err)
+	}
+
+	fileNameParts := strings.Split(inputRulesFile, ".")
+	if len(fileNameParts) != 2 {
+		return fmt.Errorf("missing filename extension or using '.' in filename")
+	}
+
+	testFile := fmt.Sprintf("%s%s%s_test.%s", dir, string(os.PathSeparator), fileNameParts[0], fileNameParts[1])
+	err = os.WriteFile(testFile, testFileContent, 0644)
+	if err != nil {
+		return fmt.Errorf("error writing rule groups test file %v", err)
+	}
+
+	cmd := exec.Command("promtool", "test", "rules", testFile)
```

Comment:
It's unfortunate that they put everything into `package main`. The drive behind having options as input and non-`main` packages is that you can just import stuff like https://github.com/prometheus/prometheus/blob/main/cmd/promtool/unittest.go#L52 and use it inline.